crates/flycheck/src/lib.rs

```diff
@@ -163,6 +174,7 @@ struct FlycheckActor {
     /// Either the workspace root of the workspace we are flychecking,
     /// or the project root of the project.
     root: AbsPathBuf,
+    state: OnceCell<FlycheckState>,
```

Comment:
What's the reason for having this?

Reply:
It solves a cancellation-related bug: a flycheck would be started, canceled before completion, and restarted. The restarted flycheck would then not interpolate the `$saved_file` value.

I _think_ with some deeper changes to Flycheck, this can be replaced by a `last_saved_file`, but this introduction of statefulness one reason I have some misgivings about this approach.